README.md

```diff
@@ -51,3 +51,110 @@ at your option.
+
+### pwm-demo (Digital Pins)
+
+The pulse width modulation demo. This demonstrates various PWM use cases by allowing the user to press buttons to change demo modes. This outputs PWM signals to the built in LEDs on the nRF52840-DK.
+
+### qdec-demo (Sensor Decoding)
+
+The quadrature decoder (QDEC) demo. This peripheral supports buffered decoding of quadrature-encoded sensor signals (typically used for mechanical and optical sensors). The demo shows how to use a rotary encoder to trigger an interrupt and update a variable by the amount of rotation. (nRF52840 + Bourns PEC11R rotary encoder)
+
+### rtic-demo (Concurrency Framework)
+
+The Real-Time Interrupt-driven Concurrency framework demo. Many of the demos in this project use RTIC and demonstrate its capabilities in more detail but this is a bare-bones default template for you to build off. RTIC is not unique to the nRF series but very useful for a program that requires concurrency. Unfortunately, this demo does not appear to use rtt for logging so it crashes when used with probe-run. However, it will work with other debuggers. See other demos for rtt logging setup.
+
+### spi-demo (Digital Pins)
+
+The serial peripheral interface master (SPIM) with EasyDMA demo. Sends some text out on the SPI peripheral and loops it back on itself to demonstrate full duplex direct-memory-access based SPI data transfer. You'll need a resistor to connect the output to the input. 
+
+### twi-ssd1306 (Digital Pins)
+
+I2C compatible Two-Wire Interface with the SSD1306 OLED Display demo. This demo uses the TWIM (Two-Wire Interface Master) peripheral along with the embedded_graphics library to draw "Hello Rust!" to an OLED screen. Note that you need to set a default feature to get this to compile (see "Running the demos" section).
+
+### twim-demo (Digital Pins)
+
+I2C compatible Two-Wire Interface Master mode demo. This demo uses the TWIM peripheral to read and write 8 bytes of data to arbitrary addresses on a device that is connected to the I2C pins p0_30 and p0_31. It demonstrates error handling if the device does not respond properly (or it is not connected).
+
+### twis-demo (Digital Pins)
+
+I2C compatible Two-Wire Interface Slave mode demo. This demo uses the twis peripheral with rtic to listen for I2C signals which are exposed as events. The event handler reads data from the peripheral at the address specified.
+
+### wdt-demo (Timer)
+
+Watchdog timer demo. This demonstrates the how the entire device can be set to automatically reset if certain conditions are not met within a certain period of time. In this case you have to press all 4 buttons at least once within a 5 second period to prevent a reset. If you mash the buttons for a while you will encounter an 'attempt to subtract with overflow' panic at `main.rs:205` which is ultimately cleared by the watchdog timer. This is intended demo behaviour ;)
```

Comment:
I think the individual demo descriptions should go into the demo's directory. That keeps all information related to a demo in a single directory and makes it easier to update.

Reply:
Thanks, I’ll make that change when I’m back at my desk on Tuesday. Strange thing was that cargo got all confused when I put a dedicated README.md file in the examples folder. I’ll experiment.

Reply:
Ok I've moved the demo related docs to its own file. Cargo was not confused, I was.

Reply:
I meant that we should put a small readme into *each* demo, which would avoid the CI issues. That way, all information for a demo is located in its directory instead of being scattered.

Reply:
Oh, I see. Some of the demos already have their own readme files. I wanted to add a quick reference guide targeted at developers unfamiliar with all the acronyms. Something to skim over when looking for a feature. For example. I was specifically looking for a bluetooth example (for which there are currently no dedicated examples) and I had to go through each demo to find it. I also didn't know how to run the demos and it's probably wasefull to repeat that information for each demo.

Reply:
Yeah the "how to run" info is fine to just put in the main readme once.

Reply:
Ok I will go ahead and create readme files for all the demos. Can I at least list the demos (and what their acronyms mean) in the main demo readme file? Things like ccm, comp, ecb, ppi, etc may be familiar to some but not to others.

Reply:
Yeah that's fine. I just don't want to put all the details there since they might change.

Reply:
Ok fair enough, cheers.

Reply:
I've moved the bulk of the text to the relevant readme files for each demo and left a short title for each demo in their place in the form of a table in the readme file in the root of the examples folder. Should I move this detail back to the readme in the root of the project itself and delete the readme in the examples folder or leave it as-is?

Reply:
Hi, I reread your comments and figured that you would have wanted me to add the detail to the main readme instead of creating a new one in the examples folder. I've gone ahead and done this. Sorry about the confusion.

Reply:
> 
> 
> Yeah that's fine. I just don't want to put all the details there since they might change.

Hey I'm not sure if you have unsubscribed from this thread. Just a reminder to check out the latest if you have. I'd like to contribute more to this repo and picked up [#60](url) which is marked as a good first issue. I'd like to help out with code reviews if you have a backlog of PRs to work though and feel pressed for time. Or some other github issue, whatever helps.

David